.vib/mongodb-sharded/ginkgo/mongodb_test.go

```diff
@@ -0,0 +1,125 @@
+package mongodb_sharded_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	utils "github.com/bitnami/charts/.vib/common-tests/ginkgo-utils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	PollingInterval = 1 * time.Second
+)
+
+var _ = Describe("MongoDB Sharded", Ordered, func() {
+	var c *kubernetes.Clientset
+	var ctx context.Context
+	var cancel context.CancelFunc
+
+	BeforeEach(func() {
+		ctx, cancel = context.WithCancel(context.Background())
+
+		conf := utils.MustBuildClusterConfig(kubeconfig)
+		c = kubernetes.NewForConfigOrDie(conf)
+	})
+
+	When("a database is created and Mongodb Shards are scaled down to 0 replicas and back up", func() {
+		It("should have access to the created database", func() {
+
+			getAvailableReplicas := func(ss *appsv1.StatefulSet) int32 { return ss.Status.AvailableReplicas }
+			getSucceededJobs := func(j *batchv1.Job) int32 { return j.Status.Succeeded }
+
+			for i := 0; i < shards; i++ {
+				By(fmt.Sprintf("checking all the shard %d replicas are available", i))
+				shardName := fmt.Sprintf("%s-shard%d-data", releaseName, i)
+				ss, err := c.AppsV1().StatefulSets(namespace).Get(ctx, shardName, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				Expect(ss.Status.Replicas).NotTo(BeZero())
+				origReplicas := *ss.Spec.Replicas
+				Eventually(func() (*appsv1.StatefulSet, error) {
+					return c.AppsV1().StatefulSets(namespace).Get(ctx, shardName, metav1.GetOptions{})
+				}, timeout, PollingInterval).Should(WithTransform(getAvailableReplicas, Equal(origReplicas)))
+			}
+
+			By("checking that mongos is available")
+			mongosName := fmt.Sprintf("%s-mongos", releaseName)
+			ss, err := c.AppsV1().StatefulSets(namespace).Get(ctx, mongosName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(ss.Status.Replicas).NotTo(BeZero())
+			mongosOrigReplicas := *ss.Spec.Replicas
```

Comment:
I think you can use:
```golang
			mongosOrigReplicas := ss.Spec.Replicas
```
The pointer should allow you to access the struct fields.